src/components/FilterList/SearchResults/hooks.ts

```diff
@@ -0,0 +1,18 @@
+import { useMemo } from 'react';
+
+import { useFilterListProvider } from '../hooks';
+
+export function useSearch<T>(
+  matcherFunc: (element: T, searchString: string) => boolean,
+  data: T[]
+) {
+  const { searchString } = useFilterListProvider();
+
+  // Memoizing this function so we don't need to recreate it when
+  // the renderElement function changes
+  const filteredElements = useMemo(
+    () => data.filter((e) => matcherFunc(e, searchString)),
+    [searchString, matcherFunc, data]
+  );
+  return filteredElements;
```

Comment:
```suggestion
  return useMemo(
    () => data.filter((e) => matcherFunc(e, searchString)),
    [searchString, matcherFunc, data]
  );
```

if the hook is renamed to `useSearchResults`, this variable is no longer needed for readability, since it becomes apparent that the this it returns is the results of the search :)

Reply:
Imo it's more readable to keep the variable 🤔

Reply:
If you think it adds something, keep it.  I personally just think it's redundant if the hook is renamed :)